src/afrolabs/components/kafka.clj

```diff
@@ -1336,8 +1336,38 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 
-(defn assert-topics
-  "Makes sure topics exist with the given number of partitions.
+(defn increase-topic-partitions!
+  "A common utility that takes a collection of topic names and a desired partition count.
+  Will query the topics to find topics with too few partitions and then increase them in-place.
+  Throws for topics that does not exist."
+  [admin-client topic-names partition-count]
+
+  (let [describe-result (-> (.describeTopics admin-client
+                                             topic-names)
+                            (.allTopicNames)
+                            (.get))
+
+        topics-createPartitions-parameter
+        (->> (for [topic topic-names
+                   :let [^TopicDescription topic-describe-result
+                         (get describe-result topic)
+
+                         max-partition
+                         (apply max (map #(.partition ^TopicPartitionInfo %)
+                                         (.partitions topic-describe-result)))]
+                   :when (< max-partition (dec partition-count))]
+               [topic (NewPartitions/increaseTo partition-count)])
+             (into {}))]
```

Comment:
I think this for would be cleaner as a `filter` and a `map` and as a bonus they could be `comp`ed `into` a transducer.

Reply:
looks like this now:
```clj
(into {}
              (comp (map (fn [topic-name]
                           (let [^TopicDescription topic-describe-result
                                 (get describe-result topic)

                                 max-partition
                                 (apply max (map #(.partition ^TopicPartitionInfo %)
                                                 (.partitions topic-describe-result)))]
                             [topic-name max-partition])))
                    (filter #(< (second %) (dec partition-count)))
                    (map (juxt first (constantly (NewPartitions/increaseTo partition-count)))))
              topic-names)
```

Reply:
or maybe with keep it can become this:
```clj
        (into {}
              (keep (fn [topic-name]
                      (let [^TopicDescription topic-describe-result
                            (get describe-result topic)

                            max-partition
                            (apply max (map #(.partition ^TopicPartitionInfo %)
                                            (.partitions topic-describe-result)))]
                        (when (< max-partition (dec partition-count))
                          [topic-name (NewPartitions/increaseTo partition-count)]))))
              topic-names)
```

Reply:
lol - looks like the for-loop to me :)

Reply:
All the steps for the filtering can just be done in the filter.
```clj
(into {}
      (comp (filter (fn [topic] (< (apply max #(.partition ^TopicPartitionInfo %) (.partitions (get describe-result topic)))
                                   (dec partition-count))))
            (map (fn [topic] [topic (NewPartitions/increaseTo partition-count)])))
      topic-names)
```
This can be made more readable by abstracting `max-partitions`
```clj
(defn max-partitions
  [describet-result topic]
  (apply max #(.partition ^TopicPartitionInfo %) (.partitions (get describe-result topic))))
```

Reply:
why a preference for `(comp (filter ...) (map ...))` over `(keep ...)`?

Reply:
The keep has a lot more logic in it, making it harder to figure out what it is supposed to do. With `(comp (filter ...) (map ...))` it is very clear that you are filtering based on a simple predicate, and then shaping it with a map. This allows them to do some semantic heavy lifting. The hiding both pieces inside `keeps` means the `keep` does carries very little of the semantic burden. 

I would argue that the point of moving to higher order functions is to let the semantics of that function do as much as possible to make the meaning clearer.

Using the theoretical `max-partitions` would drastically reduce my resistance to `keep` since the lambda would be much simpler, but I would still argue that `(comp (filter ...) (map ...))` better communicates what the code is doing.